utils/utils.go

```diff
@@ -111,12 +112,30 @@ func TarWithChroot(source string) (io.ReadCloser, error) {
 }
 
 // GuardedRemoveAll functions much like os.RemoveAll but
-// will not delete certain catastrophic paths.
+// will not delete certain catastrophic paths
+// and will ignore podman.sock file.
 func GuardedRemoveAll(path string) error {
 	if path == "" || path == "/" {
 		return fmt.Errorf("refusing to recursively delete `%s`", path)
 	}
-	return os.RemoveAll(path)
+
+	exclude := "podman.sock"
+	skipped := false
+	files, _ := os.ReadDir(path)
+	for _, file := range files {
+		if file.Name() != exclude {
+			if err := os.RemoveAll(filepath.Join(path, file.Name())); err != nil {
+				return err
+			}
+		} else {
+			skipped = true
+		}
+	}
+
+	if !skipped {
+		return os.RemoveAll(path)
+	}
+	return nil
```

Comment:
This feels like it should not be part of GuardedRemoveAll() as this seems like a more generic version. And other callers have no use of this special code.

Also this is overly complex, the skipped bool is not needed at all as you already RemoveAll() inside the loop.